js/hole-ng.tsx

```diff
@@ -98,15 +101,12 @@ async function update() {
 
 // Switch scoring
 for (const tab of scoringTabs)
-    tab.onclick = e => { e.preventDefault(); scoring = tab.id; update() };
+    tab.onclick = (e: MouseEvent) => { e.preventDefault(); scoring = tab.id; update() };
 
 // Switch lang
 const switchLang = onhashchange = () => {
-    lang = location.hash.slice(1) || localStorage.getItem('lang');
-
     // Kick 'em to Python if we don't know the chosen language.
-    if (!langs[lang])
```

Comment:
we seem to have lost the validation, that is bad.

Reply:
Ah, I misread that as `if (!lang)` and used `|| "python"` on the previous line to avoid `lang` being typed as `string | undefined`